core/src/main/java/oracle/weblogic/deploy/create/RCURunner.java

```diff
@@ -174,6 +176,56 @@ public static RCURunner createAtpRunner(String domainType, String oracleHome, St
         runner.atpTemporaryTablespace = get(rcuProperties, "atp.temp.tablespace");
         return runner;
     }
+    /**
+     * Build an RCU runner for an SSL database.
+     *
+     * @param domainType the domain type
+     * @param oracleHome the ORACLE_HOME location
+     * @param javaHome   the JAVA_HOME location
+     * @param rcuDb The URL of the database
+     * @param rcuPrefix The prefix used for the tablespaces
+     * @param rcuSchemas the list of RCU schemas to create (this list should not include STB)
+     * @param rcuVariables a comma separated list of key=value variables
+     * @param rcuProperties dictionary of SSL specific arguments
+     * @throws CreateException if a parameter validation error occurs
+     */
+    public static RCURunner createSslRunner(String domainType, String oracleHome, String javaHome, String rcuDb,
+                                            String rcuPrefix, List<String> rcuSchemas, String rcuVariables,
+                                            PyDictionary rcuProperties) throws CreateException {
+
+        String tnsAdmin = get(rcuProperties, "oracle.net.tns_admin");
+
+        RCURunner runner = new RCURunner(domainType, oracleHome, javaHome, rcuDb, rcuPrefix, rcuSchemas, rcuVariables);
+        String trustStorePassword = get(rcuProperties, "javax.net.ssl.trustStorePassword");
+        String trustStore = get(rcuProperties, "javax.net.ssl.keyStore");
+        String trustStoreType = get(rcuProperties, "javax.net.ssl.keyStoreType");
+        String keyStorePassword = get(rcuProperties, "javax.net.ssl.keyStorePassword");
+        String keyStore = get(rcuProperties, "javax.net.ssl.keyStore");
+        String keyStoreType = get(rcuProperties, "javax.net.ssl.keyStoreType");
+
+        StringBuffer sslArgs = new StringBuffer();
+        sslArgs.append("oracle.net.tns_admin=");
+        sslArgs.append(tnsAdmin);
+
+        sslArgs.append(",javax.net.ssl.trustStore=");
+        sslArgs.append(tnsAdmin + "/" + trustStore);
+        sslArgs.append(",javax.net.ssl.trustStoreType=" + trustStoreType);
+        // If wallet type is SSO, no password present
+        if (trustStorePassword != null && trustStorePassword != "None") {
+            sslArgs.append(",javax.net.ssl.trustStorePassword="+ trustStorePassword);
+        }
+        sslArgs.append(",javax.net.ssl.keyStore=");
+        sslArgs.append(tnsAdmin + "/" + keyStore);
+        sslArgs.append(",javax.net.ssl.keyStoreType=" + keyStoreType);
+        if (keyStorePassword != null && keyStorePassword != "None") {
+            sslArgs.append(",javax.net.ssl.keyStorePassword="+ keyStorePassword);
+        }
+        sslArgs.append(",oracle.net.ssl_server_dn_match=false");
```

Comment:
do we need to make ssl_server_dn_match configurable ?

Reply:
It was in a different branch and was already checked into the documentation main.

Reply:
I'm not sure we do, from what I have seen everywhere, it is false.

Reply:
Yes, I know in ATP we hard coded to false and probably we need to make it configurable - just in case user complains against it (most internal users have not much experience anyway, I don't know what it is until I read the doc) -  we can make it default to false?   I think this is an extra level of security to make sure talking to the right db server.  

From the documentation,

https://www.oracle.com/technetwork/database/enterprise-edition/wp-oracle-jdbc-thin-ssl-130128.pdf

The following property also needs to be used to force the JDBC Thin driver to verify the server’s DN:
props.setProperty("oracle.net.ssl_server_dn_match", "true");

Reply:
documentation was in another branch and it got merged already into documentation.

Reply:
from what I read online it should always be false. I might be wrong.

Reply:
Most example doesn't want to have the hassle for user to handle error.  The property exists for a reason, even though user may usually set it to false,  why not provide a option for user?

https://blogs.oracle.com/developers/post/ssl-connection-to-oracle-db-using-jdbc-tlsv12-jks-or-oracle-wallets-122-and-lower

https://docs.oracle.com/en/middleware/fusion-middleware/data-integrator/12.2.1.4/odi-marketplace/known-issues-and-workarounds.html

Reply:
Note: I am saying setting it to true, I am just asking to make it a configurable property

Reply:
sorry, I replied to your first comment twice. changing to a property